packages/charts/src/chart_types/metric/renderer/dom/text.tsx

```diff
@@ -221,19 +234,25 @@ export const MetricText: React.FunctionComponent<{
         )}
       </div>
       <div>
-        <p className="echMetricText__value" style={{ fontSize: `${VALUE_FONT_SIZE[size]}px` }}>
-          {isFiniteNumber(value)
-            ? parts.map(([type, text], i) =>
-                type === 'numeric' ? (
-                  text
-                ) : (
-                  // eslint-disable-next-line react/no-array-index-key
-                  <span key={i} className="echMetricText__part" style={{ fontSize: `${VALUE_PART_FONT_SIZE[size]}px` }}>
-                    {text}
-                  </span>
-                ),
-              )
-            : style.nonFiniteText}
+        <p className="echMetricText__value" style={{ fontSize: `${VALUE_FONT_SIZE[size]}px`, textOverflow: isMetricWNumber(datum) ? undefined : 'ellipsis'}}>
+          {isMetricWNumber(datum)
+            ? isFiniteNumber(value)
+              ? splitNumericSuffixPrefix(datum.valueFormatter(value)).map(([type, text], i) =>
+                  type === 'numeric' ? (
+                    text
+                  ) : (
+                    // eslint-disable-next-line react/no-array-index-key
+                    <span
+                      key={`${text}${i}`}
+                      className="echMetricText__part"
+                      style={{ fontSize: `${VALUE_PART_FONT_SIZE[size]}px` }}
+                    >
+                      {text}
+                    </span>
```

Comment:
Add the `title` prop to the value, assuming the `text` here is the full text.

![Screen Recording 2022-09-14 at 08 53 34 AM](https://user-images.githubusercontent.com/19007109/190203846-1d587a61-64c7-4d1f-819a-e2fcb2d6c2f5.gif)


```suggestion
                    <span
                      key={`${text}${i}`}
                      title={text}
                      className="echMetricText__part"
                      style={{ fontSize: `${VALUE_PART_FONT_SIZE[size]}px` }}
                    >
                      {text}
                    </span>
```

Reply:
great, done here: https://github.com/elastic/elastic-charts/pull/1817/commits/0ab47281cc271d6816370d1b7e60897e4857d42b
I've also cleaned up a bit the code